frame/staking/src/mock.rs

```diff
@@ -232,6 +235,89 @@ impl OnUnbalanced<NegativeImbalanceOf<Test>> for RewardRemainderMock {
 	}
 }
 
+pub struct AutomaticValidatorUpdatePerEra;
```

Comment:
So to avoid code duplication, you could put the original implementation of this in `pallet_staking`: 

```
// pallet_staking
struct SomeOpinionatedAutomaticValidatorUpdatePerEra<X, Y>; 
impl Convert<_, _> for SomeOpinionatedAutomaticValidatorUpdatePerEra {}

// node/runtime/src/lib.rs
parameter_types! { X = ..., Y = ...} 
impl staking::Config for Runtime { type AutomaticValidatorUpdatePerEra =  pallet_staking::SomeOpinionatedAutomaticValidatorUpdatePerEra<X, Y> }

```

so essentially, pallet_stkaing is exposing an optional, opinionated, and configurable implementation of this convert that may or may not be used by a runtime. 

Anf finally, then you can avoid re-writing the code here :D

Reply:
Yeah I was wondering about how to do this 👍  ... but for first iteration I wanted to see something works ;)